src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -0,0 +1,235 @@
+/**
+ * @author        Abhishek Saran
+ * @author        Manas Sanjay
+ * @module        Infrastructure module 
+ * @team          Test Harness
+ * @description   Test Harness main class
+ * @summary       To provide functionality to tester to run test case sets in various ways- 
+                  Methods such as runAll(), runByName(), runByCategroy() and run by  Priority() are provided. 
+ */
+package infrastructure.validation.testing;
+import java.io.File;
+import java.util.ArrayList;
+
+
+
+public class TestHarness{
+  
+  private String savePath;
+  
+  void setSavePath(String path){
+    this.savePath=path;
+  }
+  
+  String getSavePath(){
+    return this.savePath;
+  }
+
+   
+
+  
+  /* 
+  * Manas Sanjay may start righting his two methods runByCategroy() and run by Priority() from here. 
+  * The guideline to avoid merge conflicts is as follows-
+  * The spacing must follow a tab being two spaces otherwise identation may get messy with tabs + spaces combination.
+  * If one is using file editor gedit, then tab width has to be 2 spaces with use spaces option enabled. 
+  * Please don't edit anything beyond the space alloted to you otherwise we might get into a messy merge conflict.
+  * 
+  * For your two methods and their javadocs, 90 lines of code space(line no 38 - line no 127) is alloted to you. 
+  * In case A) you need more lines to be alloted or B) there is any need to make some changes outside the alloted space, 
+  * A pull request has to be made on my branch.
+  * I'll merge your branch and A) modify my branch to make requested changes or B) give some more lines of code space in my branch. 
+  * You need to then fetch my branch and again make a pull request on my branch for final merge.
+  */
+  //Manas's code starts from here.
+
+
+
+
+  void runByCategroy(String Category){
+    /* Parameters: Category -> The name of the category of tests to be run
+                 : Tests    -> The List of tests for which the sorting by category of tests should be done in. This function will be applicable to all the tests mentioned in the lists}
+                 */
```

Comment:
Javadoc is done by starting with **/ and ending with */. @param is used for each parameter. Each parameter is written on a single line. Just write @param {parameter_name} : {parameter_description}. Keep the description short Refer [this](https://google.github.io/styleguide/javaguide.html#s7-javadoc) on how to write javadoc